sdk/iot/hub/src/az_iot_hub_client_c2d.c

```diff
@@ -59,17 +61,18 @@ AZ_NODISCARD az_result az_iot_hub_client_c2d_parse_received_topic(
   (void)client;
 
   az_span token;
-  token = _az_span_token(received_topic, c2d_topic_suffix, &received_topic);
-  if (az_span_ptr(received_topic) != NULL)
-  {
-    token = _az_span_token(received_topic, c2d_topic_suffix, &received_topic);
-    AZ_RETURN_IF_FAILED(
-        az_iot_hub_client_properties_init(&out_request->properties, token, az_span_size(token)));
-  }
-  else
+  az_span reminder;
+  token = _az_span_token(received_topic, c2d_topic_suffix, &reminder);
```

Comment:
nit: Any reason the declaration isn't just inline with the call?
```suggestion
  az_span token = _az_span_token(received_topic, c2d_topic_suffix, &reminder);
```

Reply:
I don't think so. I'll give that a try.